metrics-influxdb/src/main/java/com/codehale/metrics/influxdb/InfluxDbHttpSender.java

```diff
@@ -0,0 +1,117 @@
+package com.codehale.metrics.influxdb;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.codec.binary.Base64;
+
+import com.codehale.metrics.influxdb.data.InfluxDbPoint;
+import com.codehale.metrics.influxdb.data.InfluxDbWriteObject;
+import com.codehale.metrics.influxdb.utils.InfluxDbWriteObjectSerializer;
+
+/**
+ * An implementation of InfluxDbSender that writes to InfluxDb via http.
+ */
+public class InfluxDbHttpSender implements InfluxDbSender {
+    private static final Charset UTF_8 = StandardCharsets.UTF_8;
+    private final URL url;
+    // The base64 encoded authString.
+    private final String authStringEncoded;
+    private final InfluxDbWriteObject influxDbWriteObject;
+    private final InfluxDbWriteObjectSerializer influxDbWriteObjectSerializer;
+
+    /**
+     * Creates a new http sender given connection details.
+     *
+     * @param hostname   the influxDb hostname
+     * @param port       the influxDb http port
+     * @param database   the influxDb database to write to
+     * @param authString the authorization string to be used to connect to InfluxDb, of format username:password
+     * @throws Exception exception while creating the influxDb sender(MalformedURLException)
+     */
+    public InfluxDbHttpSender(final String hostname, final int port, final String database, final String authString) throws Exception {
+        this(hostname, port, database, authString, TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Creates a new http sender given connection details.
+     *
+     * @param hostname      the influxDb hostname
+     * @param port          the influxDb http port
+     * @param database      the influxDb database to write to
+     * @param authString    the authorization string to be used to connect to InfluxDb, of format username:password
+     * @param timePrecision the time precision of the metrics
+     * @throws Exception exception while creating the influxDb sender(MalformedURLException)
+     */
+    public InfluxDbHttpSender(final String hostname, final int port, final String database, final String authString,
+                              final TimeUnit timePrecision) throws Exception {
+        this.url = new URL("http", hostname, port, "/write");
```

Comment:
We should either add a flag to support HTTPS or we should offer a constructor allowing to pass in the full URL.